src/WinRT.Runtime2/InteropServices/ExceptionErrorInfo.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using WindowsRuntime.InteropServices;
+using WindowsRuntime.InteropServices.Marshalling;
+
+namespace ABI.WinRT.Interop;
+
+internal static class IRestrictedErrorInfoMethods
+{
+    public static unsafe void GetErrorDetails(
+        void* thisPtr,
+        out string description,
+        out int error,
+        out string restrictedDescription,
+        out string capabilitySid)
+    {
+        void* _description = null;
+        void* _restrictedDescription = null;
+        void* _capabilitySid = null;
+
+        try
+        {
+            fixed (int* pError = &error)
+            {
+                // GetErrorDetails
+                Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<void*, void**, int*, void**, void**, int>)(*(void***)thisPtr)[3])(
```

Comment:
Let's not have inline function pointer invocations. Add a vtable type for all the interfaces we need in this PR in the `Vtables` folder. There's a whole bunch of them already there, you can follow the same exact pattern 🙂